rq/job.py

```diff
@@ -561,9 +570,26 @@ def delete(self, pipeline=None, remove_from_queue=True):
                                             job_class=self.__class__)
             failed_queue.remove(self, pipeline=pipeline)
 
+        if delete_dependents:
+            self.delete_dependents(pipeline=pipeline,
+                                   remove_from_queue=remove_from_queue)
+
         connection.delete(self.key)
         connection.delete(self.dependents_key)
 
+    def delete_dependents(self, pipeline=None, remove_from_queue=True):
```

Comment:
One more detail, we don't need the `remove_from_queue` argument because dependent jobs are not yet enqueued.